diff-test/core/README.md

```diff
@@ -0,0 +1,115 @@
+
+## Scope
+
+### Tested (Unchecked means that work is in progress. Checked means the work is complete.)
+
+The following aspects of the system are tested
+
+- [x] Sending VSC packets from provider to one consumer
+- [x] Sending VSC maturities from one consumer to provider
+- [x] The bond-based-consumer-voting-power property
+- [x] Slashing logic (not including actual token burning)
+- [x] Validator power change
+- [x] Validators leaving or joining the active validor set
+- [x] Consumer initiated slashing
+- [x] Delegation operations
+- [x] Undelegation operations
+- [x] Validator unbonding
+- [x] Valiator jailing
+- [x] Validator tombstoning
+- [x] Packet acknowledgements
+- [ ] Redelegation operations
+- [ ] Unjailing operations
+
+### NOT Tested
+
+The following aspects of the system are not tested by this work.
+
+- Completing the IBC handshakes
+- Repairing an expired IBC channel through governance
+- Slashing with non-zero slash factors
+- Submitting proposals
+- Executing proposals
+- Adding a new consumer chain
+- Removing a consumer chain for any reason
+- Distribution of rewards
+- Provider Governance
+- Consumer Governance/Democracy
+- Anything to do with cosmwasm
+- Client timeouts or expiry
+- Restarting any chain from exported state
+- Any logic that deals with having _more than one consumer chain_
+
+## Usage
+
+### Overview
+
+This typescript project contains code for
+
+- Modelling the aspects of the system listed under TESTED above
+- Generating and executing actions against a model system based on those aspects, in order to explore various behaviors. The actions are generated using heuristics and randomness.
+- Recording traces of executions to file
+- Choosing a set of traces in a manner convenient for testing the SUT.
+- Replaying a given existing trace against a new model instance, for debugging purposes.
+
+### Usage prerequisities
+
+```bash
+# nodejs version 16 is required.
+node --version
+# yarn package manager is required
+yarn --version
+# setup the project
+yarn install
+```
+
+### Commands
+
+There are several top level yarn project scripts which can be run via
+
+```bash
+yarn <script_name>
+```
+
+as per the `scripts` entry in [package.json](./package.json). The most important of these are
+
+```bash
+# install the project
+yarn install;
+# build in watch mode. Repeatedly build the project when the src changes
+# recommended to run in background process
+yarn build:watch
+# start main.ts - the entry point to the program
+yarn start <args>
+# test - run the tests in __tests__
+yarn test
+```
+
+The actual functionality has entrypoint in [src/main.ts](./src/main.ts). Please see the file for details. The available functionalities are
+
+```bash
+# generate traces
+yarn start gen <num minutes>
+# create a subset of traces
+yarn start subset <output file abs path> <num event instances (optional)>
+# replay a trace from a file (for debugging)
+yarn start replay <filename> <trace index> <num actions>
```

Comment:
Just an idea, maybe it could be possible to put models like this into a cli to allow for interactive exploration and verfication of the model's behavior

Reply:
I'm not sure I understand what you mean. Like submitting actions 1 by 1 via cli?